packages/backend/src/controllers/bills/bills.controller.ts

```diff
@@ -0,0 +1,191 @@
+      house: owner.house,
+      owner: owner._id,
+      due: new Date(createBillDto.due),
+      users: users,
+    };
+    const bill = await this.billStoreService.create(billModel);
+    return {
+      name: bill.name,
+      description: bill.description,
+      owner: owner.firebaseId,
+      due: createBillDto.due,
+      users: createBillDto.users,
+    };
+  }
+
+  @Put(':id')
+  @ApiOperation({ summary: 'Update a bill.' })
+  @ApiOkResponse({
+    description: 'Bill updated successfully',
+    type: BillResponseDto,
+  })
+  @ApiUnauthorizedResponse({ description: 'not the bill owner' })
+  async updateBill(
+    @Param() idObject: string,
+    @Body() updateBillDto: UpdateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+
+    if (bill.owner.valueOf() === userObject._id.valueOf()) {
+      const billDocument = await this.billStoreService.update(
+        bill._id,
+        updateBillDto,
+      );
+      return this.billUtil.covertBillDocumentToResponseDTO(
+        billDocument,
+        this.userStoreService,
+      );
+    } else
+      throw new HttpException('not the bill owner', HttpStatus.UNAUTHORIZED);
+  }
+
+  @Put(':id/payment')
+  @ApiOperation({ summary: 'Mark a user as having paid or unpaid a bill.' })
+  @ApiOkResponse({
+    description: 'Bill marked successfully',
+    type: BillResponseDto,
+  })
+  async updateBillPayment(
+    @Param() idObject: string,
+    @Body() payBillDto: PayBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+    bill.users.map((u) => {
+      if (u.id.valueOf() === userObject._id.valueOf()) {
+        u.paid = payBillDto.paid;
+        u.proof = payBillDto.proof;
+        console.log(u);
+      }
+    });
+    //TODO: Fix the bug of not returning the paid and proof values
+    return this.billUtil.covertBillDocumentToResponseDTO(
+      await this.billStoreService.update(bill._id, bill),
+      this.userStoreService,
+    );
+  }
+
+  @Delete(':id')
+  @ApiOperation({ summary: 'Delete a bill.' })
+  @ApiNoContentResponse({
+    description: 'Bill deleted successfully',
+    type: BillResponseDto,
+  })
+  async deleteBill(
+    @Param() idObject: string,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+
+    if (bill.owner.valueOf() === userObject._id.valueOf()) {
+      return this.billUtil.covertBillDocumentToResponseDTO(
```

Comment:
Delete endpoint should not return a dto. It should return `204` status code with no content.